### PR TITLE
[mlir][memref] Fold subview into nd-tensor descriptor

### PR DESCRIPTION
**Pull request by adam-smnk**

### CODE REVIEW

**Comment by adam-smnk** [comment]
FYI @chencha3

**Review by ftynse — CHANGES_REQUESTED** [review]

File: mlir/lib/Dialect/MemRef/Transforms/FoldMemRefAliasOps.cpp [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include "mlir/Dialect/NVGPU/IR/NVGPUDialect.h"
 #include "mlir/Dialect/Utils/IndexingUtils.h"
 #include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/Dialect/XeGPU/IR/XeGPU.h"
```
[the remark this answers is not in the document]
Comment (adam-smnk) [reply mlir/lib/Dialect/MemRef/Transforms/FoldMemRefAliasOps.cpp:25]: Fair enough. It looked like a common place to gather these folders given that NVGPU is also present here.

**Comment by adam-smnk** [comment]
To be moved into a separate XeGPU transform pass.

**Review by chencha3 — COMMENTED** [review]
Thanks @adam-smnk, LGTM.

**Pull request closed without merging**